capstone/src/main/java/com/google/step/servlets/RestaurantServlet.java

```diff
@@ -32,43 +32,48 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet responsible for sending/getting restaurants to/from Datastore. */
+/** Servlet responsible for sending/getting a restaurant to/from Datastore. */
 @WebServlet("/restaurant")
 public class RestaurantServlet extends HttpServlet {
+  /**
+   * Returns a single Restaurant based on restaurantKey.
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("RestaurantInfo").addSort("score", SortDirection.ASCENDING);
-
+    Long restaurantKeyParam = Long.parseLong(request.getParameter("restaurantKey"));
+    Query query = new Query("RestaurantInfo")
+                      .setFilter(new Query.FilterPredicate(
+                          "restaurantKey", Query.FilterOperator.EQUAL, restaurantKeyParam));
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
I forgot where we landed with this, but if it's feasible, `datastore` should really be a member of this class.

You want dependencies to be injectable, for better decoupling (and, as a side effect, it makes it possible to test).

Incidentally, you should write a unit test for this (and other) servlets, which will quickly make you hit all kinds of coupling/cohesion problems that are easy to gloss over.

For now, you can write the unit test as a separate PR later, to avoid the constantly-growing-PR effect, but after that, you should start doing smaller PRs as Megan described in today's standup.

Reply:
Changed it to be a private static member of the class!